src/otx/algo/instance_segmentation/maskrcnn_tv.py

```diff
@@ -5,140 +5,107 @@
-        for bboxes, labels, masks, polygons in zip(
-            entity.bboxes,
-            entity.labels,
-            entity.masks,
-            entity.polygons,
-        ):
-            # NOTE: shift labels by 1 as 0 is reserved for background
-            _labels = labels + 1 if len(labels) else labels
-            targets.append(
-                {
-                    "boxes": bboxes,
-                    "labels": _labels,
-                    "masks": masks,
-                    "polygons": polygons,
-                },
-            )
 
-        features = self.backbone(image_list.tensors)
-        if isinstance(features, Tensor):
-            features = OrderedDict([("0", features)])
-        proposals, proposal_losses = self.rpn(image_list, features, targets)
+class TVMaskRCNN(ExplainableOTXInstanceSegModel):
+    """Implementation of torchvision MaskRCNN for instance segmentation."""
+
+    load_from: ClassVar[dict[str, Any]] = {"maskrcnn_resnet_50": MaskRCNN_ResNet50_FPN_V2_Weights.verify("DEFAULT")}
+    mean = (123.675, 116.28, 103.53)
+    std = (58.395, 57.12, 57.375)
+
+    AVAILABLE_MODEL_VERSIONS: ClassVar[list[str]] = [
+        "maskrcnn_resnet_50",
+    ]
+
+    def _create_model(self) -> nn.Module:
+        """Create MaskRCNN model with TV implementation."""
+        if self.model_name not in self.AVAILABLE_MODEL_VERSIONS:
+            msg = f"Model version {self.model_name} is not supported."
+            raise ValueError(msg)
+
+        # NOTE: Add 1 to num_classes to account for background class.
+        num_classes = self.label_info.num_classes + 1
+        weights = self.load_from[self.model_name]
+
+        # init model components, model itself and load weights
+        rpn_anchor_generator = _default_anchorgen()
+        backbone = MaskRCNNBackbone(model_name=self.model_name)
+        rpn_head = RPNHead(model_name=self.model_name, anchorgen=rpn_anchor_generator)
+        box_head = FastRCNNConvFCHead(model_name=self.model_name)
+        mask_head = MaskRCNNHeads(model_name=self.model_name)
 
-        detections, detector_losses = self.roi_heads(
-            features,
-            proposals,
-            image_list.image_sizes,
-            targets,
+        model = MaskRCNNTV(
+            backbone,
+            num_classes=91,
+            rpn_anchor_generator=rpn_anchor_generator,
+            rpn_head=rpn_head,
+            box_head=box_head,
+            mask_head=mask_head,
         )
 
-        losses = {}
-        losses.update(detector_losses)
-        losses.update(proposal_losses)
+        model.load_state_dict(weights.get_state_dict(progress=True, check_hash=True))
 
-        if self.training:
-            return losses
-        scale_factors = [
-            img_meta.scale_factor if img_meta.scale_factor else (1.0, 1.0) for img_meta in entity.imgs_info
-        ]
+        # Replace RoIHeads since torchvision does not allow customized roi_heads.
+        model.roi_heads = TVRoIHeads(
```

Comment:
@sungchul2 RoIHeads comes from torchvision, while RoIHead is adapted from mmdetection, so they are inherently different packages. Aligning them in this PR would be challenging I think.

Reply:
Because both `TVRoIHeads` and `torchvision.RoIHeads` (set in `torchvision.FasterRCNN`) share `__init__` and have the same arguments, I roughly thought it could be seamlessly updated like this:
```python
from torchvision.models.detection.roi_heads import RoIHeads

RoIHeads = TVRoIHeads
model = MaskRCNNTV(...)
```
But I agree with that this agenda is not a scope of this PR as you said, so let's keep this.